<a href="https://colab.research.google.com/github/joeyMckinney/DS-Unit-1-Sprint-2-Statistics/blob/master/module%204/Josiah_McKinney_LS_DS_124_Introduction_to_Bayesian_Inference_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Lambda School Data Science Module 124

## Introduction to Bayesian Inference




## Assignment - Code it up!

We used pure math to apply Bayes Theorem to drug tests. Now write Python code to reproduce the results! This is purposefully open ended - you'll have to think about how you should represent probabilities and events. You can and should look things up.

Specific goals/targets:

### 1) Write a function 

`def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):` 

You should only truly need these three values in order to apply Bayes Theorem. In this example, imagine that individuals are taking a breathalyzer test with an 8% false positive rate, a 100% true positive rate, and that our prior belief about drunk driving in the population is 1/1000. 
 - What is the probability that a person is drunk after one positive breathalyzer test?
 - What is the probability that a person is drunk after two positive breathalyzer tests?
 - How many positive breathalyzer tests are needed in order to have a probability that's greater than 95% that a person is drunk beyond the legal limit?


In [16]:
def prob_drunk_given_positive(prob_drunk_prior, false_positive_rate, true_positive_rate):

  p_non_user = 1 - prob_drunk_prior
  numerator = true_positive_rate*prob_drunk_prior
  denominator = true_positive_rate*prob_drunk_prior + false_positive_rate*p_non_user

  posterior_probability = numerator/denominator
  return posterior_probability

In [17]:
posterior_probability = prob_drunk_given_positive(1/1000, .08, 1)
posterior_probability

0.012357884330202669

In [18]:
test2 = prob_drunk_given_positive(posterior_probability, .08, 1)
test2

0.13525210993291495

In [20]:
test3 = prob_drunk_given_positive(test2, .08, 1)
test3

0.6615996951348605

In [21]:
test4 = prob_drunk_given_positive(test3, .08, 1)
test4

0.9606895076105054

after one positive breathalyzer test their is a 0.012(1.2%) probability that a person is drunk

after two postives breathalyzer tests tehir is a 0.135(13.5%) probability that a person is drunk

it will take 4 postive breathalyzer tests to have a probability that is greater than 95%

### 2) In your own words, summarize the difference between Bayesian and Frequentist statistics

If you're unsure where to start, check out [this blog post of Bayes theorem with Python](https://dataconomy.com/2015/02/introduction-to-bayes-theorem-with-python/).

So frequentist statistics are more focused on the likelihood and need a bigger sample size to get more accurate results and tend to believe in their data even when they get a result that is very unlikely to happen and ignore previous results. Bayesian take prior uses results to make more accurate predictions.

### 3) Use the following Template to help come up with ideas for your Build Sprint Project: 



---

## Idea 1:
healthcare cost

### You
What do you care about?

What do you know about?

What decisions do you face?




i care about single payer healthcare system that is better than our current one

it saves money, it makes sure everyone has healthcare, its better than our current healthcare system.

what topics i need to focus on to show evidence

> Indented block





### Seven templates
Training Kit (https://learn.lambdaschool.com/ds/module/recedjanlbpqxic2r) explains the seven templates from Priceonomics.

Can you apply the templates to your topics?

yes 

**Geographic Variation:**

I can compare the cost of USA's healthcare compaired to other country's with singal payer system.

**Trend related to the news:**

their is alot of news about healthcare

**Who does that?:**

everyone who gets sick and sees a doctor uses healthcare

**Answering a question people care about:**

how much certin durgs cost compaired to other countries

**Valuable to businesses:**

if businesses didn't have to provide healthcare it would save them money.

**What's the most popular?:**

i can talk about how popular the idea is among americans.

**Cost/Money rankings:**

i can show how much USA over pays for even the most basic forms of healthcare


### Misconceptions

What misconceptions do people have about your topic?

that USA can't afford to give eveyone healthcare and it will cost more than our current system.

### Examples

What data storytelling example inspires you?

here is data that shows what health cost would look like under single payer system.
https://www.vox.com/policy-and-politics/2018/8/3/17648860/single-payer-explained-3-charts

Could you do a new hypothesis, for the same question?

yes

Could you do a new question, for the same topic?

what dose single payer look like in other city's

Could you do a new topic, with the same "style"?

yes, like our cost compaired to other contries.

###Data

Where could you search for data about your topic?

https://www.kaggle.com/shivamb/deep-healthcare-analysis-using-bigquery
https://www.kaggle.com/ravichaubey1506/healthcare-cost
https://www.kaggle.com/a1233biology/notebook


---

## Idea 2:
e-sports

### You
What do you care about?

what video game esport payers make the most money and stuff like that
how big esports are getting
how big they will get in future 

What do you know about?

what games are popular, how tournaments work, genres of video games

What decisions do you face?

what topics i want to focus on



### Seven templates
Training Kit (https://learn.lambdaschool.com/ds/module/recedjanlbpqxic2r) explains the seven templates from Priceonomics.

Can you apply the templates to your topics?

**Geographic Variation:**
how popular esports is by country or city.

**Trend related to the news:**

with corona virus outbreak esports have seen a huge spike. their is even has been F1 racing using video games.

**Who does that?:**

people who watch sports or play video games are instrested in e sports

**Answering a question people care about:**

how people currently watch esports, what do the players make, how popular is it going to get

**Valuable to businesses:**

esports is a growing business so companies could look for advertising deals based on findings about esports

**What's the most popular?:**

i think the most popular is League of Lengends for esports

**Cost/Money rankings:**

which esports teams make the most money, cost of esport teams.

### Misconceptions

What misconceptions do people have about your topic?

watching people play video games is boring.

### Examples

What data storytelling example inspires you?

https://www2.deloitte.com/us/en/insights/industry/telecommunications/capitalizing-on-growth-of-esports-industry.html
https://gdcvault.com/play/1025864/Esports-Day-Storytelling-in-Esports
https://www.esportsearnings.com/

Could you do a new hypothesis, for the same question?

yes

Could you do a new question, for the same topic?

yes

Could you do a new topic, with the same "style"?

yes


###Data

Where could you search for data about your topic?


https://www.kaggle.com/rushikeshhiray/esport-earnings
https://datarade.ai/data-categories/esports-data


## Resources

- [Worked example of Bayes rule calculation](https://en.wikipedia.org/wiki/Bayes'_theorem#Examples) (helpful as it fully breaks out the denominator)
- [Source code for mvsdist in scipy](https://github.com/scipy/scipy/blob/90534919e139d2a81c24bf08341734ff41a3db12/scipy/stats/morestats.py#L139)

## Stretch Goals:

- Go back and study the content from Modules 1 & 2 & 3 to make sure that you're really comfortable with them.
- Apply a Bayesian technique to a problem you previously worked (in an assignment or project work) on from a frequentist (standard) perspective
- Check out [PyMC3](https://docs.pymc.io/) (note this goes beyond hypothesis tests into modeling) - read the guides and work through some examples
- Take PyMC3 further - see if you can build something with it!